### DISCUSSION TITLE
Multi-select and postgreSQL

### DISCUSSION THREAD

**Discussion by NicolasFabre81** [Budibase/budibase#10828] [Help]
Hello, how can we manage the multi-select component with a postgreSQL database ?
Which data type should we use ?
How can we re-populate the field from existing data ?

I tried several things that did not work... (data type string, json ...)

Thanks for your help

**Comment by ConorWebb96** [comment]
Hey @NicolasFabre81,

I was able to get this to work with a regular text type field within my Postgres database. Initially I created it in Postgres as a text field and then amended the column once I pulled the tables into Budibase.
<img width="877" alt="Screenshot 2023-06-08 at 12 13 08" src="https://github.com/Budibase/budibase/assets/126772285/cce7aff3-7824-476b-9217-eb196b624457">

<img width="672" alt="Screenshot 2023-06-08 at 12 28 42" src="https://github.com/Budibase/budibase/assets/126772285/d607a2ec-cff0-4472-9399-2245fd32e003">

The other easier option is just to create the column in Budibase itself. It should then sync across to your external data source. This like the above will show as a text type field.

The end result of this can be seen below;
<img width="1569" alt="Screenshot 2023-06-08 at 12 32 35" src="https://github.com/Budibase/budibase/assets/126772285/a1b8a34e-00fa-4ea6-bbd1-7f9ad43795da">

The multi-select component can take it's options in 3 different ways;

- Schema, works with fields using the options column type, this is the example highlighted above.
- Custom, which is adding your own options via the add `Define options` button.
- Data provider, this lets you choose a data source and also select the label and values you want to display. I believe it will only show unique options if values are the same.

Hopefully this helps!

**Reply by NicolasFabre81** [reply]
Hello @ConorWebb96 ,
I did change the data type of my field to options like you, but I still get this error message :
<img width="157" alt="image" src="https://github.com/Budibase/budibase/assets/28592069/e696923c-86aa-40cb-aac3-985846df86be">

**Reply by ConorWebb96** [reply]
Hey @NicolasFabre81,

What version of Budibase are you using?

Could you email and app export to support@budibase.com so we can have a better look. I haven't been able to replicate this even if I attached the wrong fields.

Thanks ahead of time!

**Reply by NicolasFabre81** [reply]
I'm running V2.6.11 ! Hard to export as I'm using postgreSQL...
